incubator/hnc/test/e2e/issues_test.go

```diff
@@ -286,4 +287,21 @@ var _ = Describe("Issues that require repairing HNC", func() {
 		mustRun("kubectl delete subns", nsChild, "-n", nsParent)
 		mustNotRun("kubectl get ns", nsChild, "-o yaml")
 	})
+
+	It("Should not delete a non-leaf child namespace when deleting a parent namespace with bad anchor - issue #797", func() {
+		mustRun("kubectl hns create", nsSubChild, "-n", nsChild)
+		// Test: remove subns (anchor) in the bad parent 'parent2'
+		// Expected: The bad subns (anchor) is deleted successfully but the child is not deleted (still contains the 'test-secret')
+		mustRun("kubectl delete subns", nsChild, "-n", nsParent2)
+		mustRun("kubectl get secret -n", nsChild)
+	})
+
+	It("Should delete a non-leaf child namespace when deleting a parent namespace with good anchor under race condition - issue #797", func(){
+		mustRun("kubectl hns create", nsSubChild, "-n", nsChild)
+		// Test: Remove subns (anchor) in the good parent 'parent'
+		// Expected: The subns (anchor) is deleted successfully and the 'child' is also deleted
+		mustRun("kubectl delete subns", nsChild, "-n", nsParent)
+		mustNotRun("kubectl get ns", nsChild, "-o yaml")
+		mustNotRun("kubectl get ns", nsSubChild, "-o yaml")
```

Comment:
Hmm, it seems not right here. The original test sets the `allowCascadingDelete` first. It shouldn't delete a non-leaf subnamespace if `allowCascadingDelete` is not set.

Reply:
Thats's right.. I think I accidentally removed it when I was removing the `k apply -f`. Added back now